src/edfi-performance-test/edfi_performance_test/performance_tester.py

```diff
@@ -24,51 +24,55 @@ def prepare_for_output(args: MainArguments) -> str:
     output_base_path = f"{args.output}/pipeclean"
     return output_base_path
 
+def spawn_pref_tests(args: MainArguments) -> None:
+    # setup Environment and Runner
+    env = Environment(user_classes=[DummyUser], host=args.baseUrl)
+    env.create_local_runner()
 
-async def run(args: MainArguments) -> None:
-
-    try:
-        logger.info("Starting performance test...")
-        start = time.time()
+    # start a WebUI instance
+    env.create_web_ui("127.0.0.1", 8089)
 
-        set_config_values(args)
+    # start a greenlet that periodically outputs the current stats
 
-        # setup Environment and Runner
-        env = Environment(user_classes=[DummyUser], host=args.baseUrl)
-        env.create_local_runner()
+    stats_writer = StatsCSVFileWriter(
+        env, PERCENTILES_TO_REPORT, prepare_for_output(args), full_history=True
+    )
+    gevent.spawn(stats_writer)
 
-        # start a WebUI instance
-        env.create_web_ui("127.0.0.1", 8089)
+    # start a greenlet that save current stats to history
+    gevent.spawn(stats_history, env.runner)
 
-        # start a greenlet that periodically outputs the current stats
+    # start the test
+    if not env.runner:
+        raise RuntimeError("Locust runner not configured correctly")
+    runner = env.runner
 
-        stats_writer = StatsCSVFileWriter(
-            env, PERCENTILES_TO_REPORT, prepare_for_output(args), full_history=True
-        )
-        gevent.spawn(stats_writer)
+    runner.start(args.clientCount, spawn_rate=args.spawnRate)
 
-        # start a greenlet that save current stats to history
-        gevent.spawn(stats_history, env.runner)
+    # Manually stop locust, which otherwise would run continuously
+    gevent.spawn_later(args.runTimeInMinutes * 60, lambda: runner.quit())
 
-        # start the test
-        if not env.runner:
-            raise RuntimeError("Locust runner not configured correctly")
-        runner = env.runner
+    # wait for the greenlets
+    env.runner.greenlet.join()
 
-        runner.start(args.clientCount, spawn_rate=args.spawnRate)
+    # stop the web server for good measures
+    if env.web_ui:
+        env.web_ui.stop()
 
-        # Manually stop locust, which otherwise would run continuously
-        gevent.spawn_later(args.runTimeInMinutes * 60, lambda: runner.quit())
+async def run(args: MainArguments) -> None:
 
-        # wait for the greenlets
-        env.runner.greenlet.join()
+    try:
+        logger.info("Starting performance test...")
+        start = time.time()
 
-        # stop the web server for good measures
-        if env.web_ui:
-            env.web_ui.stop()
+        set_config_values(args)
 
-        # for running tests in a debugger
-        # run_single_user(DummyUser)
+        if(args.runInDebugMode):
+            # for running tests in a debugger
+            DummyUser.host=args.baseUrl
+            run_single_user(DummyUser)
```

Comment:
Interesting result - this function throws an error because it tries to parse Locust-style arguments and can't find `-g` or `--runInDebugMode`. So it looks like the command line arguments need to be cleared before calling this. The following solves that problem, though we also need an `import sys` statement at the top of the file.
```suggestion
            sys.argv = sys.argv[:1]
            run_single_user(DummyUser)
```